tests/integrationv2/test_fragmentation.py

```diff
@@ -61,6 +64,8 @@ def test_s2n_server_low_latency(managed_process, multi_cipher, provider, protoco
 @pytest.mark.parametrize("certificate", [Certificates.RSA_4096_SHA384, Certificates.ECDSA_384], ids=get_parameter_name)
 @pytest.mark.parametrize("frag_len", [512, 2048, 8192, 12345, 16384], ids=get_parameter_name)
 def test_s2n_server_framented_data(managed_process, multi_cipher, provider, protocol, frag_len, certificate):
+    if not provider.supports_max_frag():
```

Comment:
It depends on the version of OpenSSL being used. One of your earlier comments was to move these types of checks *into* the provider.

Reply:
The combo of "is it openssl" and "what version of openssl is it" should live in the provider, and now does.

But test-specific checks like "does the provider support max frag" still probably belong in the tests.

Reply:
I think we agree here? The test asks the provider if it supports max frag. Right now this test is only against openssl, but it will be expanded to Wolf, mbed, boring, and we shouldn't have special cases for each provider in those situations inside the test.

Reply:
We're special casing each provider for max frag no matter what. The question is just whether that special casing logic should live in the Provider or in the max frag test. I'm leaning towards the max frag test, because that's the only place that will need it.

I'm leery of the providers getting too big and cluttered :\

Reply:
Updates.